src/ledger/balances.ts

```diff
@@ -1,85 +1,61 @@
-import { Client } from "..";
-import { Connection } from "../client";
-import { validate, ensureClassicAddress } from "../common";
-import { FormattedTrustline } from "../common/types/objects/trustlines";
-
-import { GetTrustlinesOptions } from "./trustlines";
-import * as utils from "./utils";
-
-export interface Balance {
+import _ from "lodash";
+
+import { Client, dropsToXrp } from "..";
+import { LedgerIndex } from "../models/common";
+import { AccountInfoRequest } from "../models/methods";
+import {
+  AccountLinesRequest,
+  AccountLinesResponse,
+  Trustline,
+} from "../models/methods/accountLines";
+
+interface Balance {
   value: string;
   currency: string;
-  counterparty?: string;
+  issuer?: string;
 }
 
-export type GetBalances = Balance[];
-
-function getTrustlineBalanceAmount(trustline: FormattedTrustline): Balance {
-  return {
-    currency: trustline.specification.currency,
-    counterparty: trustline.specification.counterparty,
-    value: trustline.state.balance,
-  };
+function formatBalances(trustlines: Trustline[]): Balance[] {
+  return trustlines.map((trustline) => ({
+    value: trustline.balance,
+    currency: trustline.currency,
+    issuer: trustline.account,
+  }));
 }
 
-function formatBalances(
-  options: GetTrustlinesOptions,
-  balances: { xrp: string; trustlines: FormattedTrustline[] }
-) {
-  const result = balances.trustlines.map(getTrustlineBalanceAmount);
-  if (
-    !(options.counterparty || (options.currency && options.currency !== "XRP"))
-  ) {
-    const xrpBalance = {
-      currency: "XRP",
-      value: balances.xrp,
-    };
-    result.unshift(xrpBalance);
-  }
-  if (options.limit && result.length > options.limit) {
-    const toRemove = result.length - options.limit;
-    result.splice(-toRemove, toRemove);
-  }
-  return result;
-}
-
-function getLedgerVersionHelper(
-  connection: Connection,
-  optionValue?: number
-): Promise<number> {
-  if (optionValue != null && optionValue !== null) {
-    return Promise.resolve(optionValue);
-  }
-  return connection
-    .request({
-      command: "ledger",
-      ledger_index: "validated",
-    })
-    .then((response) => response.result.ledger_index);
-}
-
-function getBalances(
+async function getBalances(
   this: Client,
```

Comment:
```suggestion
  client: Client,
```
It's not good form to use `this` when you're not in the actual class